import-scripts/dmp-import-vars-functions.sh

```diff
@@ -312,82 +319,46 @@ function filter_derived_clinical_data {
-    echo $(date)
-    TOMCAT_HOST_LIST=(dashi.cbio.mskcc.org dashi2.cbio.mskcc.org)
-    TOMCAT_HOST_USERNAME=cbioportal_importer
-    TOMCAT_HOST_SSH_KEY_FILE=${HOME}/.ssh/id_rsa_schultz_tomcat_restarts_key
-    TOMCAT_SERVER_RESTART_PATH=/srv/data/portal-cron/schultz-tomcat-restart
-    TOMCAT_SERVER_PRETTY_DISPLAY_NAME="Schultz Tomcat" # e.g. Public Tomcat
-    TOMCAT_SERVER_DISPLAY_NAME="schultz-tomcat" # e.g. schultz-tomcat
-    SSH_OPTIONS="-i ${TOMCAT_HOST_SSH_KEY_FILE} -o BATCHMODE=yes -o ConnectTimeout=3"
-    declare -a failed_restart_server_list
-    for server in ${TOMCAT_HOST_LIST[@]} ; do
-        if ! ssh ${SSH_OPTIONS} ${TOMCAT_HOST_USERNAME}@${server} touch ${TOMCAT_SERVER_RESTART_PATH} ; then
-            failed_restart_server_list[${#failed_restart_server_list[*]}]=${server}
-        fi
-    done
-    if [ ${#failed_restart_server_list[*]} -ne 0 ] ; then
-        EMAIL_BODY="Attempt to trigger a restart of the $TOMCAT_SERVER_DISPLAY_NAME server on the following hosts failed: ${failed_restart_server_list[*]}"
-        echo -e "Sending email $EMAIL_BODY"
-        echo -e "$EMAIL_BODY" | mail -s "$TOMCAT_SERVER_PRETTY_DISPLAY_NAME Restart Error : unable to trigger restart" $PIPELINES_EMAIL_LIST
-    fi
+function clearPersistenceCachesForMskPortals() {
+    all_msk_portals="msk msk-beta"
+    clearPersistenceCachesForPortals "$all_msk_portals"
 }
 
-# Function for restarting Triage tomcats
-# TODO obviously restartMSKTomcats and restartSchultzTomcats and restartTriageTomcats should really be one function ...
-function restartTriageTomcats {
-    # redeploy war
-    echo "Requesting redeployment of triage portal war..."
-    echo $(date)
-    TOMCAT_HOST_LIST=(dashi.cbio.mskcc.org dashi2.cbio.mskcc.org)
-    TOMCAT_HOST_USERNAME=cbioportal_importer
-    TOMCAT_HOST_SSH_KEY_FILE=${HOME}/.ssh/id_rsa_triage_tomcat_restarts_key
-    TOMCAT_SERVER_RESTART_PATH=/srv/data/portal-cron/restart-trigger/triage-tomcat-restart
-    TOMCAT_SERVER_PRETTY_DISPLAY_NAME="Triage Tomcat" # e.g. Public Tomcat
-    TOMCAT_SERVER_DISPLAY_NAME="triage-tomcat" # e.g. triage-tomcat
-    SSH_OPTIONS="-i ${TOMCAT_HOST_SSH_KEY_FILE} -o BATCHMODE=yes -o ConnectTimeout=3"
-    declare -a failed_restart_server_list
-    for server in ${TOMCAT_HOST_LIST[@]} ; do
-        if ! ssh ${SSH_OPTIONS} ${TOMCAT_HOST_USERNAME}@${server} "umask 000 ; touch ${TOMCAT_SERVER_RESTART_PATH}" ; then
-            failed_restart_server_list[${#failed_restart_server_list[*]}]=${server}
-        fi
-    done
-    if [ ${#failed_restart_server_list[*]} -ne 0 ] ; then
-        EMAIL_BODY="Attempt to trigger a restart of the $TOMCAT_SERVER_DISPLAY_NAME server on the following hosts failed: ${failed_restart_server_list[*]}"
-        echo -e "Sending email $EMAIL_BODY"
-        echo -e "$EMAIL_BODY" | mail -s "$TOMCAT_SERVER_PRETTY_DISPLAY_NAME Restart Error : unable to trigger restart" $PIPELINES_EMAIL_LIST
-    fi
+function clearPersistenceCachesForExternalPortals() {
```

Comment:
Considering that these portals share the same db and all get refreshed together at once - maybe we just switch the deployments to use the same redis db and clear that single one? 

Would save both on the memory being used and the time restarting each cache

Reply:
I like this idea ... the clearCache script has temporary URLs listed anyway and will need a subsequent update/PR. I suggest that we experiment with the deployments and try out this suggestion. When the deployment configuration has stabilized we can adjust the scripts so that the appropriate website URLs are called for clearing the cache. If only one external portal redis is running, we can cut down the number of URLs which are hit.